build.rs

```diff
@@ -182,28 +212,18 @@ fn main() -> Result<(), Box<dyn Error>> {
         .flag("-Wno-unused-command-line-argument")
         .clone();
 
-    enum Device {
-        NanoS,
-        NanoSPlus,
-        NanoX,
-    }
-    use Device::*;
-
-    // determine device
-    let device = match env::var_os("CARGO_CFG_TARGET_OS").unwrap().to_str().unwrap() {
-        "nanos" => NanoS,
-        "nanosplus" => NanoSPlus,
-        "nanox" => NanoX,
-        target_name => panic!(
-            "invalid target `{}`, expected one of `nanos`, `nanox`, `nanosplus`. Run with `-Z build-std=core --target=./<target name>.json`",
-            target_name
-        ),
-    };
+    // Determine which device we're compiling for
+    let device = Device::detect()?;
+
+    // Export `NANOS_SDK_TARGET` and `NANOS_SDK_ROOT` for consuming crates
+    println!("cargo:rustc-env=TARGET={}", device);
```

Comment:
I don't understand why this non-standard stuff is needed? What is wrong with `target_os = nanos` in CFGs?

Reply:
maybe there's another way of achieving this but i wanted to be able to read `NANOS_SDK_TARGET` from a build script in another package

Reply:
My understanding was that `CARGO_CFG_TARGET_OS`, as the standardized env var, will be set by Cargo in every package, so you can just use that?